test/integration/optimistic.test.js

```diff
@@ -0,0 +1,382 @@
+    {
+        id: 'b645cf572a8f5acf8716e4846b408d3b1ca45c58',
+        source: '"use strict";module.exports.world=function(){return"world"};',
+        deps: {},
+        file: './assets/js/foo.js',
+    },
+];
+
+const feed3 = [
+    {
+        id: 'a645cf572a8f5acf8716e4846b408d3b1ca45c58',
+        source: '"use strict";module.exports.world=function(){return"world"};',
+        deps: {},
+        file: './assets/js/baz.js',
+    },
+];
+
+const feed4 = [
+    {
+        id: 'e645cf572a8f5acf8716e4846b408d3b1ca45c58',
+        source: '"use strict";module.exports.world=function(){return"world"};',
+        deps: {},
+        file: './assets/js/foobar.js',
+    },
+];
+
+const cssFeed1 = [
+    {
+        id: 'c645cf572a8f5acf8716e4846b408d3b1ca45c58',
+        source: 'h1 { color: red; }',
+        deps: {},
+        file: './assets/css/bar.js',
+    },
+];
+
+test('optimistic bundling of js feeds', async () => {
+    const sink = new Sink();
+    const optimisticBundler = new OptimisticBundler({ sink });
+
+    await optimisticBundler.publishAssets({
+        tag: 'podlet1',
+        type: 'js',
+        data: feed1,
+    });
+    await optimisticBundler.publishInstructions({
+        tag: 'layout1',
+        type: 'js',
+        data: ['podlet1'],
+    });
+
+    expect(
+        await sink.get(
+            '8b9a3142fa84751f4a0b6578f58024bd3b98aea6fb47f8e87dd0c813a250fa95.js'
+        )
+    ).toMatchSnapshot();
+});
+
+test('publish instructions before publishing assets', async () => {
+    const sink = new Sink();
+    const optimisticBundler = new OptimisticBundler({ sink });
+
+    await optimisticBundler.publishInstructions({
+        tag: 'layout1',
+        type: 'js',
+        data: ['podlet1'],
+    });
+
+    await optimisticBundler.publishAssets({
+        tag: 'podlet1',
+        type: 'js',
+        data: feed1,
+    });
+
+    expect(
+        await sink.get(
+            '8b9a3142fa84751f4a0b6578f58024bd3b98aea6fb47f8e87dd0c813a250fa95.js'
+        )
+    ).toMatchSnapshot();
+});
+
+test('optimistic bundling of css feeds', async () => {
+    const sink = new Sink();
+    const optimisticBundler = new OptimisticBundler({ sink });
+
+    await optimisticBundler.publishAssets({
+        tag: 'podlet1',
+        type: 'css',
+        data: cssFeed1,
+    });
+
+    await optimisticBundler.publishInstructions({
+        tag: 'layout1',
+        type: 'css',
+        data: ['podlet1'],
+    });
+
+    expect(
+        await sink.get(
+            '2c408796d55d03d6fbaef774c591b868cf07cac4147bc61d3b40825e16f725ab.css'
+        )
+    ).toMatchSnapshot();
+});
+
+test('publish instructions before publishing muiltple assets', async () => {
+    const sink = new Sink();
+    const optimisticBundler = new OptimisticBundler({ sink });
+
+    await optimisticBundler.publishInstructions({
+        tag: 'layout1',
+        type: 'js',
+        data: ['podlet1', 'podlet2', 'podlet3'],
+    });
+
+    await optimisticBundler.publishAssets({
```

Comment:
can we `publishAssets` in parallel to speed up the tests a tiny bit?

Reply:
Interestingly, after some measuring, it seems that Promise.all is slower than running those operations one after another. 

Those tests use the memory sink so likely without decently slow IO there's just no win in parallelization.